test/e2e/performanceprofile/functests/utils/node_inspector/inspector.go

```diff
@@ -19,58 +19,91 @@ import (
 	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
 	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/daemonset"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/images"
 	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/namespaces"
 	testpods "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
 )
 
 const serviceAccountSuffix = "sa"
 const clusterRoleSuffix = "cr"
 const clusterRoleBindingSuffix = "crb"
 
-func Create(cli client.Client, namespace, name, image string) error {
-	serviceAccountName := fmt.Sprintf("%s-%s", name, serviceAccountSuffix)
-	sa := createServiceAccount(serviceAccountName, namespace)
-	if err := cli.Create(context.Background(), sa); err != nil && !errors.IsAlreadyExists(err) {
+var initialized bool
+var namespace *corev1.Namespace = namespaces.NodeInspectorNamespace
+var nodeInspectorName = testutils.NodeInspectorName
+
+// initialize would be used to lazy initialize the node inspector
+func initialize() error {
+	if initialized {
+		return nil
+	}
+	// Create the test namespace
+	err := testclient.DataPlaneClient.Create(context.TODO(), namespace)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failed to create namespace: %v", err)
+	}
+
+	// Create Node Inspector resources
+	err = create()
+	if err != nil {
+		return fmt.Errorf("Failed to create Node Inspector resources: %v", err)
+	}
+
+	return nil
+}
+
+func create() error {
+	serviceAccountName := fmt.Sprintf("%s-%s", nodeInspectorName, serviceAccountSuffix)
+	sa := createServiceAccount(serviceAccountName, namespace.Name)
+	if err := testclient.DataPlaneClient.Create(context.Background(), sa); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
If you're using `context.Background()` pass the same one to all the calls

Reply:
even better would be to add a context to `create` and let the caller set it.